### ISSUE TITLE
New scripts not watched

### ISSUE THREAD

**Issue by aaafly** [jakemmarsh/angularjs-gulp-browserify-boilerplate#41]
This issue seems related to https://github.com/jakemmarsh/angularjs-gulp-browserify-boilerplate/issues/12

When creating or renaming a controller file, I see the lint task running but nothing else.
It doesn't rebundle as expected.

If I add browserify in the watch.js then it's working.

```
gulp.watch(config.scripts.src, ['lint', 'browserify']);
```

My Environment:
Windows 8.1
node v0.12.2
npm 2.7.4
gulp 3.8.11

**Comment by rwam** [comment]
Same problem here, but if you add browserify task again it ends with a performance issue. See my comment on https://github.com/jakemmarsh/angularjs-gulp-browserify-boilerplate/issues/12#issuecomment-75732981 I think it's a browserify or watchify issue like https://github.com/substack/watchify/issues/56.

**Comment by jakemmarsh** [comment]
As @rwam said, adding the `browserify` task back would result in duplicate rebundlings and performance issues because it would then be running from `watchify` still, too. This definitely sounds like [substack/watchify#56](https://github.com/substack/watchify/issues/56). However, as mentioned in that thread, if you `require` any new files in your existing files, they would be detected and rebundled.

**Comment by rwam** [comment]
No, they wouldn't ;( It's more a problem with bulk-require and mentioned on https://github.com/substack/watchify/issues/101 to describe the behavior.

**Comment by dziamid** [comment]
Is there a need to use watchify in this boilerplate? AFAIK, if we are doing a single bundle with browserify, then there's no real benefit of using watchify over gulp.watch.

**Comment by dziamid** [comment]
@rwam  This need to be automated. How about setting up gulp.watch to watch glob for new an deleted source files and do something that would make browerify to include these files into the build?

```
  gulp.watch(config.scripts.src, function (e) {

    if (e.type == 'added' || e.type == 'deleted') {
      console.log('New\deteled source file detected ', e.path);
      //TODO: do something to make browserify aware of this change
    }
  });
```

**Comment by piggyslasher** [comment]
@dziamid the current `tasks/watch.js` is like so:

```javascript
'use strict';

import config from '../config';
import gulp   from 'gulp';

gulp.task('watch', ['browserSync'], function() {

  // Scripts are automatically watched and rebundled by Watchify inside Browserify task
  gulp.watch(config.scripts.src, ['lint']);
  gulp.watch(config.styles.src,  ['styles']);
  gulp.watch(config.images.src,  ['images']);
  gulp.watch(config.fonts.src,   ['fonts']);
  gulp.watch(config.views.watch, ['views']);

});
```
